### PR TITLE
Add notebooks to documentation

### LINKED ISSUE #100: Add dataset notebooks to documentation
Add the notebooks that are currently in the main branch (`mk_data_exploration_...`) to the documentation page of 'datasets'.

### PR DESCRIPTION
**Pull request by SarahAlidoost**
closes #100

This PR adds notebooks, and recipes to doc. It also reorganizes the doc. Preview the documentation [here](https://springtime--105.org.readthedocs.build/en/105/).

### CODE REVIEW

File: docs/example_usecases.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+Here, we provide several notebooks to show how to use `springtime` package in
+different use cases. Click on each title to download the notebook.
+
+## Data exploration
+
+- [NPN](notebooks/mk_data_exploration_npn.ipynb)
+- [PEP725](notebooks/mk_data_exploration_pep725.ipynb)
+- [PPO](notebooks/mk_data_exploration_ppo.ipynb)
+- [Daymet](mk_data_exploration_daymet.ipynb)
```
Comment (sverhoeven) [inline docs/example_usecases.md:15]:
```suggestion
- [Daymet](notebooks/mk_data_exploration_daymet.ipynb)
```

**Review by sverhoeven — APPROVED** [review]
All notebooks work except one, see inline suggestion.

The collapsed recipes at https://springtime--105.org.readthedocs.build/en/105/recipes/ look nice.

Why are we rendering the datasets notebooks, but downloading the example use case notebooks? Why not render them both? As you can add download link to rendered notebooks with https://github.com/danielfrg/mkdocs-jupyter#download-notebook-link

File: docs/example_usecases.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!--
+SPDX-FileCopyrightText: 2023 Springtime authors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+Here, we provide several notebooks to show how to use `springtime` package in
+different use cases. Click on each title to download the notebook.
```
Comment (Peter9192) [inline docs/example_usecases.md:8]: I'd humbly suggest to remove this page from the docs.

File: docs/recipes.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+SPDX-License-Identifier: Apache-2.0
+-->
+
+We provide several "recipes" for downloading data from various sources and
+executing a range of ML methods with it.
+Note that a "recipe" is a file with `yaml` extension.
+
+## Datasets
+
+??? example "Daymet recipe"
```
Comment (Peter9192) [inline docs/recipes.md:13]: These foldable boxes are really nice! I just wonder if they should be presented as "example use cases" or rather as part of the "datasets" documentation.

File: docs/notebooks/datasets_modis.ipynb [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{
```
Comment (Peter9192) [inline docs/notebooks/datasets_modis.ipynb:1]: Empty notebook - shall we make a new issue to add this?

File: mkdocs.yml [hunk 5]
```diff
@@ -47,13 +61,23 @@ theme:
 plugins:
 - mkdocs-jupyter:
       include_source: True
-      include: ["pk_wishful_workflows.ipynb"]
+      include: [
+        "pk_wishful_workflows.ipynb",
```
Comment (Peter9192) [inline mkdocs.yml:65]: I agree the wishful notebook can be repurposed as a mini-tutorial, but it should be completely rewritten. Shall we make a new issue for that?

File: docs/recipes.md [hunk 6]
```diff
@@ -0,0 +1,58 @@
+<!--
+SPDX-FileCopyrightText: 2023 Springtime authors
+
+SPDX-License-Identifier: Apache-2.0
+-->
+
+We provide several "recipes" for downloading data from various sources and
+executing a range of ML methods with it.
+Note that a "recipe" is a file with `yaml` extension.
```
Comment (Peter9192) [inline docs/recipes.md:9]: We're still missing a good explanation of what are/how to use the recipes vs the API. See other comment about rewriting the example use case.

**Review by Peter9192 — APPROVED** [review]
Nice work, the package already looks so much richer now 🤩 
But it also highlights a few other this that would be nice to address (probably in follow-up items):

- Some R calls are very noisy. In `rnpn` we used `--no-echo` and captured stderr, we could consider doing this for all notebooks, perhaps as part of #92
- It would be nice to add some explanation about the various datasets and their sources.

File: docs/example_usecases.md (hunk 1 above)
Comment (SarahAlidoost) [reply docs/example_usecases.md]: This page is removed, [as suggested by Peter](https://github.com/phenology/springtime/pull/105#discussion_r1154054637).

**Comment by SarahAlidoost** [comment]
> All notebooks work except one, see inline suggestion.
> 
> The collapsed recipes at https://springtime--105.org.readthedocs.build/en/105/recipes/ look nice.
> 
> Why are we rendering the datasets notebooks, but downloading the example use case notebooks? Why not render them both? As you can add download link to rendered notebooks with https://github.com/danielfrg/mkdocs-jupyter#download-notebook-link

Download links are added, see an [example](https://springtime--105.org.readthedocs.build/en/105/notebooks/datasets_npn/).

File: docs/example_usecases.md (hunk 2 above)
Comment (SarahAlidoost) [reply docs/example_usecases.md]: removed!

File: docs/recipes.md (hunk 3 above)
Comment (SarahAlidoost) [reply docs/recipes.md:11]: moved to `Datasets` page. However, the `Datasets` page re-organized in the future considering API usage and recipe usage.

File: docs/notebooks/datasets_modis.ipynb (hunk 4 above)
Comment (SarahAlidoost) [reply docs/notebooks/datasets_modis.ipynb:1]: It is filled with examples now.

File: mkdocs.yml (hunk 5 above)
Comment (SarahAlidoost) [reply mkdocs.yml]: issue #111

**Comment by SarahAlidoost** [comment]
> * It would be nice to add some explanation about the various datasets and their sources.
@Peter9192 each notebook has a link to the data source. See an [example](https://springtime--105.org.readthedocs.build/en/105/notebooks/datasets_npn/). Are more explanations about datasets needed?

File: docs/recipes.md (hunk 6 above)
Comment (SarahAlidoost) [reply docs/recipes.md:9]: I added some explanations under a new title "Usage" in the readme, see my [commit](https://github.com/phenology/springtime/pull/105/commits/df6c172404194c348ad49598ec35e53755e1a381).

**Pull request merged**